### PR TITLE
feat: Add APM environment variables to vm_server.

### LINKED ISSUE #15: Investigate using DataDog for APM
Research option to use DataDog for APM (Application performance monitoring)

### PR DESCRIPTION
**Pull request by zmmille2**
update readme and sample file.

Closes #15.

https://docs.datadoghq.com/tracing/setup_overview/setup/dotnet-framework/?tab=environmentvariables

MVC 4.5+ will automatically pick up on these variables and support tracing via DataDog.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -6,9 +6,14 @@ This documentation breaks down the steps to deploy the .net framework app (Hello
...
 
+Make sure to store your backend variables in a `backend.tfvars` file based off [the sample backend.tfvars file](./terraform/backend.tfvars.sample).
+You'll also need to have a `terraform.tfvars` file based off [the saample terraform.tfvars file](./terraform/terraform.tfvars.sample).
+
 ```text
+cd terraform
+terraform init -backend-config backend.tfvars
 terraform workspace new dev
-terraform init
+terraform workspace select dev
```
Comment (tessmichi) [inline Readme.md:16]: hm, still need there to be an init at least for the first run

File: terraform/vm_server.tf [hunk 2]
```diff
@@ -216,3 +216,18 @@ resource "azurerm_security_center_assessment" "vmSCA" {
     code = "Healthy"
   }
 }
+
+resource "azurerm_virtual_machine_extension" "vm_script_extension" {
```
Comment (tessmichi) [inline terraform/vm_server.tf:220]: want to make sure @jding05  sees this as this is one of a few absolutely valid ways to run a script on each vm being made

**Review by tessmichi — APPROVED** [review]

File: Readme.md (hunk 1 above)
Comment (wbreza) [reply Readme.md:16]: `init` command is on line 14.

File: terraform/backend.tfvars.sample [hunk 3]
```diff
@@ -0,0 +1,4 @@
+resource_group_name = "Name of the resource group which stores remote state."
```
Comment (wbreza) [inline terraform/backend.tfvars.sample:1]: nit: Jeff already has a PR out that has this change.

File: terraform/vm_server.tf [hunk 4]
```diff
@@ -216,3 +216,18 @@ resource "azurerm_security_center_assessment" "vmSCA" {
...
+  name                 = "hostname"
+  virtual_machine_id   = azurerm_windows_virtual_machine.vm_server.id
+  publisher            = "Microsoft.Azure.Extensions"
+  type                 = "CustomScript"
+  type_handler_version = "2.0"
+
+  # https://docs.datadoghq.com/tracing/setup_overview/setup/dotnet-framework/?tab=environmentvariables#windows-services
+  settings = <<SETTINGS
+    {
+        "commandToExecute": "SET COR_ENABLE_PROFILING=1 && SET COR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}"
```
Comment (wbreza) [inline terraform/vm_server.tf:230]: I'm assuming these are the env vars that datadog uses to setup all the eventing?

**Review by wbreza — APPROVED** [review]
LGTM

File: terraform/vm_server.tf (hunk 4 above)
Comment (zmmille2) [reply terraform/vm_server.tf:230]: Yes - I put the comment above linking that documentation. Happy to make this more clear if I can.

File: terraform/vm_server.tf (hunk 2 above)
Comment (zmmille2) [reply terraform/vm_server.tf:220]: @jding05 Please ack!

**Review by jding05 — APPROVED** [review]

**Pull request merged**